Logging Refactor - Remove Print Statements

basilwong opened the pull request:
Summary:
Removes print statements and implements logging via the logging library.

Hopefully this will allow more control on the level of logging when running models.

Test Plan:
```
AOT_PARTITIONER_DEBUG=1 buck2 run @mode/opt //aps_models/ads/icvr:icvr_launcher -- mode=local_fb_fm_v4 launcher.num_workers=2
```

Resulting output paste: P1674535630
* Full logs paste: P1674535621

```
pastry P1674535621 | grep "functorch/partitioners.py" | pastry
```

Logging results: P1674549514

Differential Revision: D61678215

Skylion007 commented on torch/_functorch/partitioners.py:
```diff
@@ -824,8 +824,7 @@ def solve_min_cut(
         ops_ignored = joint_module_ops - {str(i) for i in op_types.recomputable_ops}
-        print("Ops banned from re-materialization: ", ops_ignored)
-        print()
+        logger.info("Ops banned from re-materialization: ", ops_ignored)
```
```suggestion
        logger.info("Ops banned from re-materialization: %s", ops_ignored)
```
Unlike print, second args will be ignored otherwise

Skylion007 commented on torch/_functorch/partitioners.py:
```diff
@@ -1923,14 +1922,14 @@ def classify_nodes(joint_module):
-        print(
+        sorted_sizes = sorted([(_size_of(i), str(i)) for i in saved_values])
+        logger.debug(
             "Theoretical Activations Stored: ",
```
```suggestion
            "Theoretical Activations Stored: %f",
```

Skylion007 commented on torch/_functorch/partitioners.py:
```diff
@@ -1923,14 +1922,14 @@ def classify_nodes(joint_module):
         )
-        sorted_sizes = sorted([(_size_of(i), str(i)) for i in saved_values])
+        logger.debug(
+            f"Theoretical Per Activation Storage Sizes: {sorted_sizes}"
```
Ruff will complain about this in a second

Skylion007 replied:
```suggestion
            "Theoretical Per Activation Storage Sizes: %s", {sorted_sizes}
```

zeshengzong commented on torch/_functorch/partitioners.py:
```diff
@@ -1943,10 +1942,10 @@ def classify_nodes(joint_module):
         )
-        print(
+        logger.debug(
             "Count of Ops Rematerialized: ",
```
```suggestion
            "Count of Ops Rematerialized: %s",
```

Skylion007 commented:
Check the ruff errors in lintrunner, it will complain for you for a lot of this logging.

Skylion007 commented on torch/_functorch/partitioners.py:
```diff
@@ -510,7 +510,7 @@ def _count_ops(graph: fx.Graph):
         if node.op == "call_function":
             cnt[node.target.__name__] += 1
-    print(sorted(cnt.items(), key=lambda x: x[1], reverse=True))
+    logger.info(sorted(cnt.items(), key=lambda x: x[1], reverse=True))
```
```suggestion
    logger.info("%s", sorted(cnt.items(), key=lambda x: x[1], reverse=True)))
```

basilwong commented:
Updated the diff based on comments. Testing also confirms the output.

basilwong commented:
@pytorchbot label "topic: not user facing"

basilwong commented:
/easycla

basilwong commented:
@Skylion007 able to stamp this pr?

Ran `lintrunner` on the code so it should align with code base requirements.

jansel commented on torch/_functorch/partitioners.py:
```diff
@@ -37,8 +37,8 @@
-AOT_PARTITIONER_DEBUG = config.debug_partitioner
-log = logging.getLogger(__name__)
+AOT_PARTITIONER_DEBUG: bool = config.debug_partitioner
+logger: logging.Logger = logging.getLogger(__name__)
```
Let's leave this named `log` rather than `logger` since that is the convention elsewhere.

basilwong replied:
Updated the PR with this change.

huydhn commented:
@pytorchbot rebase

basilwong committed:
Logging Refactor - Remove Print Statements (#139782)

Summary:

Removes print statements and implements logging via the logging library.

Hopefully this will allow more control on the level of logging when running models.

Test Plan:
```
AOT_PARTITIONER_DEBUG=1 buck2 run mode/opt //aps_models/ads/icvr:icvr_launcher -- mode=local_fb_fm_v4 launcher.num_workers=2
```

Resulting output paste: P1674535630
* Full logs paste: P1674535621

```
pastry P1674535621 | grep "functorch/partitioners.py" | pastry
```

Logging results: P1674549514

Reviewed By: paryxyt

Differential Revision: D61678215
